server/src/streaming/systems/users.rs

```diff
@@ -200,29 +256,49 @@ impl System {
             );
             return Err(Error::InvalidCredentials);
         }
-        if let Some(client_id) = client_id {
-            let mut client_manager = self.client_manager.write().await;
-            client_manager.set_user_id(client_id, user.id).await?;
-        }
+
         info!("Logged in user: {username} with ID: {}.", user.id);
+        if session.is_none() {
+            return Ok(user);
+        }
+
+        let session = session.unwrap();
+        if session.is_authenticated() {
+            warn!(
+                "User with ID: {} was already authenticated, removing the previous session...",
+                session.user_id
+            );
+            self.logout_user(session).await?;
+        }
+
+        session.set_user_id(user.id);
+        let mut client_manager = self.client_manager.write().await;
+        client_manager
+            .set_user_id(session.client_id, user.id)
+            .await?;
         Ok(user)
     }
 
-    pub async fn logout_user(&self, user_id: u32, client_id: Option<u32>) -> Result<(), Error> {
-        if user_id == 0 {
+    pub async fn logout_user(&self, session: &Session) -> Result<(), Error> {
+        if !session.is_authenticated() || session.user_id == 0 {
             warn!("Cannot logout the unauthenticated user (you're probably running the server without enabled authentication).");
             return Ok(());
```

Comment:
passive verb sounds better in logs: `server is presumably running without authentication`

Reply:
Fixed :)